logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -12,7 +12,8 @@ def initialize(logger, output_class, metric, strategy_registry, plugin_args)
     @output_class = output_class
     @metric = metric
     @plugin_args = plugin_args
-    @strategy_registry ||= ::LogStash::OutputDelegatorStrategyRegistry.instance
+    @strategy_registry = strategy_registry
```

Comment:
is this still an issue even though there should be consistent scoping for such a variable?

```ruby
class Test
  attr_reader :foo

  def initialize(foo)
    @foo = foo
    if foo
      puts "OK"
    else
      puts "NOT OK"
    end
  end
end

test = Test.new("foo")
```

prints:

```
"OK"
```